atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/TableRendererV2.java

```diff
@@ -0,0 +1,422 @@
+                            SortedMap.class, RowResult.class, Persistables.class)
+                    .addStatement("return results\n"
+                                    + ".values()\n"
+                                    + ".stream()\n"
+                                    + ".map(entry -> $T.of(entry).get$N())\n"
+                                    + ".collect($T.toList())",
+                            rowResultType, VarName(col), Collectors.class);
+
+            return getterBuilder.build();
+        }
+
+        private MethodSpec renderNamedGetAllRowsColumn(NamedColumnDescription col) {
+            MethodSpec.Builder getterBuilder = MethodSpec.methodBuilder("getAll" + VarName(col) + 's')
+                    .addModifiers(Modifier.PUBLIC);
+
+            getterBuilder.returns(ParameterizedTypeName.get(
+                    ClassName.get(List.class),
+                    ClassName.get(col.getValue().getValueType().getTypeClass())));
+
+
+            getterBuilder
+                    .addStatement("$T colSelection = $T.create($T.singletonList($T.toCachedBytes($S)))",
+                            ColumnSelection.class, ColumnSelection.class, Collections.class,
+                            PtBytes.class, col.getShortName())
+                    .addStatement("return $T.of(t.getRange("
+                                        + "tableRef, "
+                                        + "$T.builder().retainColumns(colSelection).build()))\n"
+                                    + ".immutableCopy()\n"
+                                    + ".stream()\n"
+                                    + ".map(entry -> $T.of(entry).get$N())\n"
+                                    + ".collect($T.toList())",
+                            BatchingVisitableView.class, RangeRequest.class, rowResultType,
+                            VarName(col), Collectors.class);
+
+            return getterBuilder.build();
+        }
+
+        private MethodSpec renderNamedDeleteRow() {
+            MethodSpec.Builder deleteRowBuilder = MethodSpec.methodBuilder("deleteRow")
+                    .addModifiers(Modifier.PUBLIC);
+
+            deleteRowBuilder = addParametersFromRowComponents(deleteRowBuilder, tableMetadata);
+
+            SortedSet<NamedColumnDescription> namedColumns = ColumnRenderers.namedColumns(tableMetadata);
+            deleteRowBuilder
+                    .addStatement("$T row = $T.of($N)", rowType, rowType, getArgumentsFromRowComponents(tableMetadata))
+                    .addStatement("byte[] rowBytes = row.persistToBytes()", Persistables.class)
+                    .addStatement("$T<$T> cells = $T.of()", Set.class, Cell.class, ImmutableSet.class);
+
+            for (NamedColumnDescription col : namedColumns) {
+                deleteRowBuilder.addStatement("cells.add($T.create(rowBytes, $T.toCachedBytes($N)))",
+                        Cell.class, PtBytes.class, ColumnRenderers.short_name(col));
```

Comment:
do we want these classes to substituted via `$T` ?

Reply:
Yeah, that's how Javapoet figures out that they need to be imported.